docs/_static/C++/CMakeLists.txt

```diff
@@ -0,0 +1,129 @@
+cmake_minimum_required(VERSION 3.0)
+
+project(TestVisualizer)
+
+find_package(PkgConfig QUIET)
+find_package(OpenGL    QUIET)
+find_package(pybind11  QUIET)
+find_package(OpenMP    QUIET)
+find_package(Open3D    QUIET HINTS "~/open3d_install/lib/CMake")
+
+if (PKGCONFIG_FOUND)
+    pkg_search_module(EIGEN3    eigen3>=3.2.7   QUIET)
+    pkg_search_module(JPEG      libjpeg         QUIET)
+    pkg_search_module(GLFW      glfw3           QUIET)
+    pkg_search_module(GLEW      glew            QUIET)
+    pkg_search_module(JSONCPP   jsoncpp>=1.7.0  QUIET)
+    pkg_search_module(PNG       libpng>=1.6.0   QUIET)
+endif (PKGCONFIG_FOUND)
```

Comment:
These lines force CMake to find the packages installed in the system.
However, I just made a quick test, and there is an easy workaround to build everything from source code included in Open3D. In this way, you don't even need to install the libraries, 😁 

First, you need to apply the change I mentioned above for GLFW CMakeLists.txt file. Then once you install Open3D, all header files will be copied to Open3D include folder and all libraries will be copied to Open3D lib folder.

Then create a CMakeLists.txt like this:
```
cmake_minimum_required(VERSION 3.0)

project(TestVisualizer)

find_package(PkgConfig QUIET)
find_package(OpenGL    QUIET)
find_package(pybind11  QUIET)
find_package(OpenMP    QUIET)
find_package(Open3D    QUIET HINTS "~/open3d_install/lib/CMake")

# Open3D
if (Open3D_FOUND)
    message(STATUS "Found Open3D ${Open3D_VERSION}")

    add_executable(TestVisualizer TestVisualizer.cpp)

    include_directories(${Open3D_INCLUDE_DIRS})

    link_directories(${Open3D_LIBRARY_DIRS})

    set_target_properties(TestVisualizer PROPERTIES
            RUNTIME_OUTPUT_DIRECTORY "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")

    target_link_libraries(TestVisualizer ${Open3D_LIBRARIES} jsoncpp jpeg glew png zlib glfw3 opengl32 glu32)
    target_link_libraries(TestVisualizer tinyfiledialogs)

else ()
    message(SEND_ERROR "Open3D not found")
endif ()
```

I have tested it on Windows 10 + VS 2017. It works. Actually, the right way to do it is: when we compile Open3D and install it, we write a customized FindOpen3D.cmake file. It remembers which dependent libraries are compiled from source included in Open3D, and include these dependent libraries in ``${Open3D_LIBRARIES}``. In this case, when an external C++ program is linked to ``${Open3D_LIBRARIES}``, it links to the Open3D library and all dependent libraries that were compiled with it.

Reply:
At this time I recommend we move fwd with this PR as is (with only minor modifications) and address the build dependencies issue in a separate PR.

There actually multiple problems, I'll mention below just two of the major ones.
I can fix these however this will take more than just half a day.
1. GLFW_LIBRARIES is set to glfw but the output of the build GLFW is actually glfw3. Somehow this works fine when building Open3D but when building a custom C++ project only based on installed Open3D libs & headers (and dependencies) the linker fails to find the lib. Setting the GLFW_LIBRARIES to glfw3 breaks the build of Open3D. Setting the Open3D_LIBRARIES to "...; glfw3;..." still doesn't work as the custom build project expects more libraries (dependencies of GLFW). I tested this on Ubuntu after uninstalling _libglfw3_ and _libglfw3-dev_.

```
[100%] Linking CXX executable TestVisualizer
/usr/bin/ld: /home/dpetre/open3d_install/lib/libglfw3.a(vulkan.c.o): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
//lib/x86_64-linux-gnu/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
CMakeFiles/TestVisualizer.dir/build.make:94: recipe for target 'TestVisualizer' failed
make[2]: *** [TestVisualizer] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/TestVisualizer.dir/all' failed
make[1]: *** [CMakeFiles/TestVisualizer.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

2. set(GLFW_INSTALL ON) breaks the `make uninstall`

```
CMake Error at cmake_uninstall.cmake:3 (message):
  Cannot find install manifest:
  "/home/dpetre/open3d/issue_130/build/External/GLFW/install_manifest.txt"

External/GLFW/CMakeFiles/uninstall.dir/build.make:57: recipe for target 'External/GLFW/CMakeFiles/uninstall' failed
make[3]: *** [External/GLFW/CMakeFiles/uninstall] Error 1
CMakeFiles/Makefile2:226: recipe for target 'External/GLFW/CMakeFiles/uninstall.dir/all' failed
make[2]: *** [External/GLFW/CMakeFiles/uninstall.dir/all] Error 2
CMakeFiles/Makefile2:233: recipe for target 'External/GLFW/CMakeFiles/uninstall.dir/rule' failed
make[1]: *** [External/GLFW/CMakeFiles/uninstall.dir/rule] Error 2
Makefile:188: recipe for target 'uninstall' failed
make: *** [uninstall] Error 2
```

Reply:
otherwise all the other mechanisms (like custom creating Open3D include dirs and libraries based on build options) mentioned by QianYi are already in place.